pandas/_libs/index.pyx

```diff
@@ -368,7 +375,9 @@ cdef class IndexEngine:
                     # GH#35392
                     if need_nan_check:
                         # Do this check only once
-                        stargets_has_nan = any(util.is_nan(val) for x in stargets)
+                        stargets_has_nan = any(
```

Comment:
think this is worth fixing in a separate PR?  we should find a test case where fixing this makes a difference

Reply:
don't think this will ever cause a user facing issue. this line will unnecessarily update `d` when `np.nan` is present in the Index (even if `np.nan` isn't the target):

```
import numpy as np
from pandas import Index
import pandas._testing as tm

index = Index(["b", "b", np.nan, np.nan])
# assertion statement passes, but d = {'b': [0, 1], nan: [2, 3]}
indexer, missing = index.get_indexer_non_unique(["b"])
expected_indexer = np.array([0, 1], dtype=np.intp)
expected_missing = np.array([], dtype=np.intp)
tm.assert_numpy_array_equal(indexer, expected_indexer)
tm.assert_numpy_array_equal(missing, expected_missing)
```

do you have any recommendation on how to write a test for a case like this? ie.  is there a way we can write a test to check for the correct value of `d` within `get_indexer_non_unique`?

Reply:
best guess, will look more closely later: try using `np.nan, float('nan')` instead of using `np.nan, np.nan`